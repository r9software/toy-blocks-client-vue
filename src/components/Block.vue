<template>
 <v-row class="container">
   <v-col class="subtitle-text" cols="12">
     {{block.attributes.index}}
   </v-col>
   <v-col class="description-text" cols="12">
     {{block.attributes.data}}
   </v-col>
 </v-row>
</template>

<script>
export default {
name: "Block",
  props:['block'],
}
</script>

<style scoped>
  .container{
    background-color:#e0e0e0;
    margin-left: 0 !important;
    margin-right: 0 !important;
    margin-bottom: 6px;
    padding:0;
  }
  .subtitle-text{
    color: #2f4ffe;
    font-size: 12px;
    padding-top: 6px;
    padding-bottom: 0px;
    padding-left: 8px;
    font-weight: 500;
    line-height: 14px;
    letter-spacing: 0.44px;
  }
  .description-text{
    color: black;
    font-size: 14px;
    padding-top: 6px;
    padding-left: 8px;
    line-height: 14px;
    letter-spacing: 0.44px;
  }

</style>