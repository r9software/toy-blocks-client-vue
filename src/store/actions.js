export const actions = {
  async getAllNodes({ commit }, nodeList) {
    for(const el of nodeList) {
      await commit('checkNodeStatusStart', el);
      
      try {
        const res = await fetch(`${el.url}/api/v1/status`);
        const response = await res.json();
        const params = {
          el,
          name: response.node_name
        };
        await commit('checkNodeStatusSuccess', params);
      }
      catch (e) {
        await commit('checkNodeStatusFailure', el);
      }
    }
  },
  async getAllBlocksForNode({commit},node) {
      try {
        const res = await fetch(`${node.url}/api/v1/blocks`);
        const response = await res.json();
        const params = {
          el: node,
          blocks:response.data
        };
        console.log(params)
        await commit('nodeBlocksResponseSuccess', params);
      }
      catch (e) {
        await commit('nodeBlocksResponseFailure', {el: node});
      }
  },

}