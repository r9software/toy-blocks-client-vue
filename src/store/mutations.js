export const mutations = {
  checkNodeStatusStart(state, { url }) {
    let list = state.nodes.list;
    const index = state.nodes.list.findIndex(t => t.url === url);
    if (index >= 0) {
      list = [
        ...state.nodes.list.slice(0, index),
        {
          ...state.nodes.list[index],
          loading: true
        },
        ...state.nodes.list.slice(index + 1)
      ];
    }
    state.nodes = {
      list: [
        ...list,
      ]
    }
  },

  checkNodeStatusSuccess(state, { el : { url }, name }) {
    let list = state.nodes.list;
    const index = state.nodes.list.findIndex(t => t.url === url);
    if (index >= 0) {
      list = [
        ...state.nodes.list.slice(0, index),
        {
          ...state.nodes.list[index],
          name: name,
          online: true,
          loading: false,
          blocks:[],
        },
        ...state.nodes.list.slice(index + 1)
      ];
    }

    state.nodes = {
      list: [
        ...list,
      ]
    }
  },
  nodeBlocksResponseSuccess(state,{ el : { url }, blocks }) {
    let list = state.nodes.list;
    const index = state.nodes.list.findIndex(t => t.url === url);
    if (index >= 0) {
      list = [
        ...state.nodes.list.slice(0, index),
        {
          ...state.nodes.list[index],
          blocks:blocks
        },
        ...state.nodes.list.slice(index + 1)
      ];
    }

    state.nodes = {
      list: [
        ...list,
      ]
    }
  },
  nodeBlocksResponseFailure(state, { el : { url } }) {
    let list = state.nodes.list;
    const index = state.nodes.list.findIndex(t => t.url === url);
    if (index >= 0) {
      list = [
        ...state.nodes.list.slice(0, index),
        {
          ...state.nodes.list[index],
          blocks:[],
        },
        ...state.nodes.list.slice(index + 1)
      ];
    }

    state.nodes = {
      list: [
        ...list,
      ]
    }
  },
  checkNodeStatusFailure(state, { url }) {
    let list = state.nodes.list;
    const index = state.nodes.list.findIndex(t => t.url === url);
    if (index >= 0) {
      list = [
        ...state.nodes.list.slice(0, index),
        {
          ...state.nodes.list[index],
          online: false,
          loading: false,
          blocks:[],
        },
        ...state.nodes.list.slice(index + 1)
      ];
    }

    state.nodes = {
      list: [
        ...list,
      ]
    }
  },
}