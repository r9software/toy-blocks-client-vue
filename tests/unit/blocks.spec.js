import Vue from 'vue';
import { shallowMount } from '@vue/test-utils'
import Node from '@/components/Node.vue'
import Block from '@/components/Block.vue'
import store from '../../src/store'
import Vuetify from 'vuetify'

Vue.use(Vuetify)

describe('<Block />', () => {
  const defaultProps = {};

  //mocks fetch API
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve({ el: { url: 'https://thawing-springs-53971.herokuapp.com' }, node_name: 'Test Name' }),
    })
  );

  const render = (props = {node:{
      url: "",
      online: true,
      name: "sample-node",
      loading: false,
      blocks:[{attributes:{index:0,data:"simple-text"}}],
    }}) => shallowMount(Node, {
    store,
    comments:{Block},
    propsData: {
      ...defaultProps,
      ...props
    }
  });
  
  it('renders', () => {
    const wrapper = render();
    expect(wrapper.find('.expansion-container').exists()).toBeTruthy;
  });

  it('match the snapshot', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders nodes', () => {
    const wrapper = render();
    expect(wrapper.findComponent(Node).exists()).toBeTruthy;
  });
});