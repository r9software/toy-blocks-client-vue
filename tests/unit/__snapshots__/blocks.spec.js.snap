// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> match the snapshot 1`] = `
<v-expansion-panel-stub class="accordion-node">
  <v-expansion-panel-header-stub expandicon="$expand">
    <div class="d-flex">
      <div class="d-flex flex-column"><span class="accordion-header-name">sample-node</span> <span class="accordion-header-url"></span></div>
      <v-badge-stub color="#18cc55" left="true" value="true" transition="scale-rotate-transition" bordered="true" dot="true" label="$vuetify.badge" inline="true" class="accordion-badge"><span class="text-uppercase accordion-status-text">Online</span></v-badge-stub>
    </div>
  </v-expansion-panel-header-stub>
  <v-expansion-panel-content-stub class="expansion-container">
    <block-stub block="[object Object]"></block-stub>
  </v-expansion-panel-content-stub>
</v-expansion-panel-stub>
`;
